yt/docker/ytsaurus/Dockerfile

```diff
@@ -228,3 +258,75 @@ FROM scratch AS ytsaurus-server-override
 COPY --from=ytsaurus-server-override-bloated / /
 
 ##########################################################################################
+
+FROM python-base as odin-installer
+
+ARG ODIN_ROOT
+ARG ODIN_VIRTUAL_ENV
+ARG ODIN_CHECKS_DIR
+ARG ODIN_CHECKS_DATA_DIR
+
+ARG SOURCE_ROOT="${ODIN_ROOT}/ytsaurus"
+ARG ODIN_SOURCE_ROOT="${SOURCE_ROOT}/yt/odin"
+ARG BUILD_ROOT="${ODIN_ROOT}/build"
+ARG PYTHON_ROOT="${ODIN_ROOT}/python"
+
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends \
+      ca-certificates \
+      curl \
+      build-essential \
+      libcap-dev \
+    && rm -rf /var/lib/apt/lists/*
+
+COPY --link ./ ${SOURCE_ROOT}/
+
+WORKDIR ${ODIN_ROOT}
+
+RUN pip install virtualenv \
+    && virtualenv ${ODIN_VIRTUAL_ENV}
+
+RUN . ${ODIN_VIRTUAL_ENV}/bin/activate \
+    && mkdir ${PYTHON_ROOT} \
+    && cd ${SOURCE_ROOT} && pip install -e yt/python/packages \
+    && cd ${PYTHON_ROOT} \
+    && python ${ODIN_SOURCE_ROOT}/packages/prepare_python_modules.py --source-root "$SOURCE_ROOT" --output-path "$PYTHON_ROOT" \
+    && cp "${ODIN_SOURCE_ROOT}/packages/setup.py" ./ && python setup.py bdist_wheel --universal
+
+RUN . ${ODIN_VIRTUAL_ENV}/bin/activate \
+    && pip install -r ${ODIN_SOURCE_ROOT}/requirements.txt \
+    && pip install ytsaurus-client ytsaurus-yson ytsaurus-rpc-driver \
+  && pip install $PYTHON_ROOT/dist/*
+
+RUN mkdir -p ${ODIN_CHECKS_DIR} \
+    && mkdir -p ${ODIN_CHECKS_DATA_DIR} \
+    && for CHECK in $(find ${ODIN_SOURCE_ROOT}/checks/bin -mindepth 1 -maxdepth 1 -type d -not -name "ya.make"); do \
```

Comment:
Maybe use here `;` instead of `&&` ?

Reply:
Why `-not -name "ya.make"` needed? It is not a directory already

Reply:
I think && is better, so we do not rely on the fact that the last statement should be erroneous in case any of the first mkdirs fail (otherwise docker would swallow the error).

As for `-not -name "ya.make"` I think it is not neccessary, removed it.